File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Resources/renderresourceusageandsheetview.rb
module Asposetasksjava
  module RenderResourceUsageAndSheetView
    def initialize()
        # Rendering Resource Usage
        render_resource_usage()

        # Rendering Resource Sheet View
        render_resource_sheetview()
    end

    def render_resource_usage()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

		# Define the SaveOptions with required TimeScale settings as Days
		options = Rjb::import('com.aspose.tasks.PdfSaveOptions').new
		options.setTimescale(Rjb::import('com.aspose.tasks.Timescale').Days)
		# Set the Presentation format to ResourceUsage
		options.setPresentationFormat(Rjb::import('com.aspose.tasks.PresentationFormat').ResourceUsage)
		# Save the Project
		project.save(data_dir + "result_days.pdf", options)

		# Set the Tiemscale settings to ThirdsOfMonths
		options.setTimescale(Rjb::import('com.aspose.tasks.Timescale').ThirdsOfMonths)
		# Save the Project
		project.save(data_dir + "result_thirdsOfMonths.pdf", options)

		# Set the Timescale settings to Months
		options.setTimescale(Rjb::import('com.aspose.tasks.Timescale').Months)
		# Save the project
		project.save(data_dir + "result_months.pdf", options)

		puts "Created resource usage files."
    end

    def render_resource_sheetview()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

		# Define the SaveOptions with required TimeScale settings as Days
		options = Rjb::import('com.aspose.tasks.PdfSaveOptions').new
		
		# Set the Presentation format to ResourceSheet
		options.setPresentationFormat(Rjb::import('com.aspose.tasks.PresentationFormat').ResourceSheet)
        project.save(data_dir + "result.pdf", options)

		puts "Created resource sheet view file."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResources/RenderResourceUsageAndSheetView.php
<?php
namespace Aspose\Tasks\WorkingWithResources;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\PdfSaveOptions as PdfSaveOptions;
use com\aspose\tasks\Timescale as Timescale;
use com\aspose\tasks\PresentationFormat as PresentationFormat;

class RenderResourceUsageAndSheetView{

    public static function run($dataDir=null){

        # Rendering Resource Usage
        RenderResourceUsageAndSheetView::render_resource_usage($dataDir);

        # Rendering Resource Sheet View
        RenderResourceUsageAndSheetView::render_resource_sheetview($dataDir);

    }

    public static function render_resource_usage($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

		# Define the SaveOptions with required TimeScale settings as Days
		$options = new PdfSaveOptions();
        $timescale=new Timescale();
        $options->setTimescale($timescale->Days);
		# Set the Presentation format to ResourceUsage
        $presentationFormat=new PresentationFormat();
		$options->setPresentationFormat($presentationFormat->ResourceUsage);
		# Save the Project
		$project->save($dataDir . "result_days.pdf", $options);

		# Set the Tiemscale settings to ThirdsOfMonths
		$options->setTimescale($timescale->ThirdsOfMonths);
		# Save the Project
		$project->save($dataDir . "result_thirdsOfMonths.pdf", $options);

		# Set the Timescale settings to Months
		$options->setTimescale($timescale->Months);
		# Save the project
		$project->save($dataDir . "result_months.pdf", $options);

		print "Created resource usage files.".PHP_EOL;

    }

    public static function render_resource_sheetview($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

		# Define the SaveOptions with required TimeScale settings as Days
		$options = new PdfSaveOptions();

        # Set the Presentation format to ResourceSheet
        $presentationFormat=new PresentationFormat();
        $options->setPresentationFormat($presentationFormat->ResourceSheet);
        $project->save($dataDir . "result.pdf", $options);
		print "Created resource sheet view file.".PHP_EOL;
    }


}
?>